tds-item-selection-components/src/main/java/tds/itemselection/services/impl/MsbAssessmentSelectionServiceImpl.java

```diff
@@ -37,6 +37,11 @@ public ItemCandidatesData selectFixedMsbSegment(UUID opportunityKey) throws Exce
       itemCandidatesService.getAllItemCandidates(opportunityKey);
     if (itemCandidates.isEmpty()) return null;
     ItemCandidatesData adaptiveSegmentData = itemCandidates.get(0);
+
+    if(adaptiveSegmentData.getAlgorithm().equalsIgnoreCase(AlgorithmType.SATISFIED.getType())) {
```

Comment:
This is a slight change on our side and msb.  First, when we get to the last segment we return a `Satisfied` algorithm type which is handled properly by the item selection service.

This wasn't working right with our version of this flow.